CHANGELOG.md

```diff
@@ -6,6 +6,25 @@ and this library adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+### Changed
+- `zcashlc_rewind_to_height` now returns an `i64` value instead of a boolean. The
+  value `-1` indicates failure; any other height indicates the height to which the
+  data store was actually truncated. Also, this procedure now takes an additional
+  `safe_rewind_ret` parameter that, on failure to rewind, will be set to the
+  minimum height for which the rewind would succeed or will remain unmodified if
+  no such height can be determined.
```

Comment:
I generally think that it's a bad idea for out parameters to only be set conditionally. It's too easy to fail to initialize them in the caller (for example, [here](https://github.com/zcash/zcash/pull/6926/files#diff-63c7710d72381cffe692175dd70e47f1891bf75c3e09249ed826f9df3917d02dL321) there was undefined behaviour due to `nFeeRequired` being uninitialized and only conditionally set). In this case I think it would be better to use an out parameter that is a `std::optional`, and set it to `std::nullopt` at the start of `zcashlc_rewind_to_height` (suggestion below).